### PR TITLE
delete pinned message announcement message

### LINKED ISSUE #921: Delete the pinned-message in help threads
We have a feature that auto-pins the first message in each help-thread.

However, we should delete the message that indicates that event in discord, as it creates unecessary noise:

pin

### PR DESCRIPTION
**Pull request by ankitsmt211**
Might not be the best implementation, but open to ideas.
* resolves #921

### CODE REVIEW

File: application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java [hunk 1]
```diff
@@ -88,6 +90,8 @@ private void handleHelpThreadCreated(ThreadChannel threadChannel) {
         // Sending messages at that moment is not allowed.
         createMessages(threadChannel).and(pinOriginalQuestion(threadChannel))
             .queueAfter(5, TimeUnit.SECONDS);
+
+        deletePinnedAnnouncement(threadChannel);
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java:94]: please put it into the same single action-queue.

File: application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java [hunk 2]
```diff
@@ -182,4 +186,17 @@ public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
         throw new UnsupportedOperationException("Not used");
     }
 
+    private void deletePinnedAnnouncement(ThreadChannel channel) {
+        channel.getHistoryFromBeginning(FIRST_TEN)
+            .map(historyFromBeginning -> historyFromBeginning.getRetrievedHistory()
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java:191]: just `history` is a better var name imo

File: application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java [hunk 3]
```diff
@@ -182,4 +186,17 @@ public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
...
+                .stream()
+                .filter(message -> message.getType() == MessageType.CHANNEL_PINNED_ADD)
+                .findFirst())
+            .flatMap(message -> message
+                .orElseThrow(() -> new IllegalStateException(
+                        "Unable to retrieve pinned message announcement"))
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java:197]: i wouldnt give this error that much attention. its possible that it fails and not necessarily a bug. for example if the thread was insta-auto-deleted bc its scam.

in such a case, this "cleanup" code shouldnt generate unecessary noise that no one is interested in. id rather log it as `log.debug(...)`

File: application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java [hunk 4]
```diff
@@ -182,4 +186,17 @@ public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
         throw new UnsupportedOperationException("Not used");
     }
 
+    private void deletePinnedAnnouncement(ThreadChannel channel) {
+        channel.getHistoryFromBeginning(FIRST_TEN)
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java:190]: @Tais993 better idea how to approach it? if not, id say its okay

**Review by Zabuzard — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java (hunk 4 above)
Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java:190]: Why not just listen to all messages, and then remove the "... pinned a message to this message, see all pinned messages". if it happens in that specific channel.

Regardless, this is some weird code, it's intentionally delaying everything with a couple of seconds.
Maybe there is some specific motivation behind that, but it looks very weird to delay 1 queue with 5s, the other with 10s.

Comment (Zabuzard) [reply application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java:190]: agreed. just making a message listener that deletes any "pinned" message in all threads in `#questions` is likely easier and cleaner

Comment (ankitsmt211) [reply application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java:190]: very well, i'll get started on this.

**Comment by ankitsmt211** [comment]
I linked a new PR based on suggested ideas, will close this one now.

**Pull request closed without merging**